app/server/appsmith-server/src/main/java/com/appsmith/server/services/NewActionServiceImpl.java

```diff
@@ -702,31 +704,48 @@ private ActionExecutionResult addDataTypesAndSetSuggestedWidget(ActionExecutionR
      * Suggest the best widget to the query response. We currently planning to support List, Select, Table and Chart widgets
      * @return
      */
-    private WidgetType getSuggestedWidget(Object data) {
+    private List<WidgetType> getSuggestedWidget(Object data) {
 
-        if(data instanceof String) {
-            return WidgetType.TEXT_WIDGET;
-        }
+        List<WidgetType> widgetTypeList = new ArrayList<>();
 
-        if(data instanceof ArrayNode && !((ArrayNode) data).isEmpty()  && ((ArrayNode) data).isArray()) {
-            ArrayNode array = (ArrayNode) data;
-            Integer length = array.size();
-            ObjectNode objectNode = (ObjectNode)array.get(0);
-            Integer fieldsCount = array.get(0).size();
-            if( (objectNode.has("x") || (objectNode.has("X")) )&&
-                    (objectNode.has("y") || (objectNode.has("Y"))) ) {
-                return WidgetType.CHART_WIDGET;
-            }
-            if(fieldsCount <= 2) {
-                return WidgetType.DROP_DOWN_WIDGET;
-            }
-            if(length <= 20 && fieldsCount <= 5) {
-                return WidgetType.LIST_WIDGET;
+        if(data instanceof ArrayNode && ((ArrayNode) data).isArray()) {
+            if(!((ArrayNode) data).isEmpty()) {
+                try {
+                    ArrayNode array = (ArrayNode) data;
+                    int length = array.size();
+                    JsonNode node = array.get(0);
+                    JsonNodeType nodeType = node.getNodeType();
+
+                    if(nodeType.equals(JsonNodeType.STRING)) {
+                        if (length > 1) {
+                            widgetTypeList.add(WidgetType.DROP_DOWN_WIDGET);
+                        }
+                        else {
+                            widgetTypeList.add(WidgetType.TEXT_WIDGET);
+                        }
+                    }
+
+                    if(nodeType.equals(JsonNodeType.OBJECT) || nodeType.equals(JsonNodeType.ARRAY)) {
+                        widgetTypeList.add(WidgetType.CHART_WIDGET);
+                        widgetTypeList.add(WidgetType.DROP_DOWN_WIDGET);
+                        widgetTypeList.add(WidgetType.LIST_WIDGET);
+                        widgetTypeList.add(WidgetType.TABLE_WIDGET);
+                    }
+
+                    if(nodeType.equals(JsonNodeType.NUMBER)) {
+                        widgetTypeList.add(WidgetType.INPUT_WIDGET);
+                        widgetTypeList.add(WidgetType.TEXT_WIDGET);
+                    }
+
+                } catch(ClassCastException e) {
+                    log.warn("Error while casting data to suggest widget "+ e);
```

Comment:
```suggestion
                    log.warn("Error while casting data to suggest widget.", e);
```
Nit: Having `,e` instead of `+e` ensures that the stack trace is also printed in the logs.